src/js/jsx/search/SearchBar.jsx

```diff
@@ -88,50 +91,188 @@ define(function (require, exports, module) {
+                if (layerItem.title === newInfo) {
+                    layerItem.displayInfo = "";
+                } else {
+                    layerItem.displayInfo = newInfo;
+                }
+            });
+
+            var layerLabel = {
+                    id: "layer_header",
+                    title: "Layers",
+                    type: "header"
+                },
                 layerOptions = layerMap.unshift(layerLabel);
 
-            // get list of currently open documents
-            var docStore = this.getFlux().store("document"),
+            return layerOptions;
+        },
+
+        /**
+         * Make list of currently open documents to be used as dropdown options
+         * 
+         * @return {Array.<Object>}
+         */
+        _getCurrDocOptions: function () {
+            var appStore = this.getFlux().store("application"),
+                docStore = this.getFlux().store("document"),
+                document = appStore.getCurrentDocument(),
                 openDocs = Immutable.fromJS(appStore.getOpenDocumentIDs()).filterNot(function (doc) {
                                 return doc === document.id;
                             }),
                 docMap = openDocs.map(function (doc) {
-                    return { id: "curr-doc_" + doc.toString(), title: docStore.getDocument(doc).name, type: "item" };
+                    return {
+                        id: "curr-doc_" + doc.toString(),
+                        title: docStore.getDocument(doc).name,
+                        type: "item"
+                    };
                 }),
-                docLabel = { id: "curr-doc_header", title: "Documents", type: "header" },
+
+                docLabel = {
+                    id: "curr-doc_header",
+                    title: "Documents",
+                    type: "header"
+                },
+
                 docOptions = docMap.unshift(docLabel);
 
-            return layerOptions.concat(docOptions);
+            return docOptions;
+        },
+
+        /**
+         * Make list of recent documents to be used as dropdown options
+         * 
+         * @return {Array.<Object>}
+         */
+        _getRecentDocOptions: function () {
+            var appStore = this.getFlux().store("application"),
+                recentFiles = appStore.getRecentFiles(),
+                recentDocMap = recentFiles.map(function (doc) {
+                    return {
+                        id: "recent-doc_" + doc,
+                        title: pathUtil.getShortestUniquePaths([doc])[0],
+                        type: "item",
+                        info: doc,
+                        displayInfo: doc
+                    };
+                });
+            
+            // Get shortest unique file path
+            var paths = collection.pluck(recentDocMap, "info"),
+                shortenedPaths = pathUtil.getShortestUniquePaths(paths);
+
+            recentDocMap.map(function (docItem, index) {
```

Comment:
I can change it to `.forEach` but I don't think it can go into the mapping above because it needs to first look have the info of all the layers to find the shortest unique path.